infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -429,20 +486,23 @@ private void handleFailures(PollReport pollReport) {
      *
      * @param pollReport Poll Report from running the failure detection policy.
      */
-    private void correctOutOfPhaseEpochs(PollReport pollReport) {
+    private void correctWrongEpochs(PollReport pollReport) {
```

Comment:
I think this needs to evolve further. If I am not too off, this is to correct LayoutServers. Using the word epoch is wrong here. In any case we are missing some important correction facility.
1. If the commit message does not reach any layoutServer, this facility has to complete that. Commit message is an optimization in our protocol( and for a good reason). But the layout is already chose as long as phase2 completes. We need to fix that.

Reply:
Ok but this cannot be left out . It needs to be done in this release.